src/test/ui/casts-issue-46365.rs

```diff
@@ -3,5 +3,5 @@ struct Lorem {
 }
 
 fn main() {
-    let _foo: *mut Lorem = 0 as *mut _; // no error here
```

Comment:
FWIW this was specifically testing `as` casts (judging from https://github.com/rust-lang/rust/issues/46365 and https://github.com/rust-lang/rust/pull/46973), so I think this change lost our intended test coverage. I am hence reverting it in https://github.com/rust-lang/rust/pull/97765.

I have not audited the other tests that were changed here.